### ISSUE TITLE
Training with own dataset

### ISSUE THREAD

**Issue by bratao** [lukemelas/EfficientNet-PyTorch#4]
Hello @lukemelas ,
I'm very excited about your code. Because I want to test in a problem I'm working with. I'm just starting with Image classifiers.

Could you please help me or provide an example where we can use our model to train a data-set from scratch?

Best regards!

**Comment by lukemelas** [comment]
I am working on code to train from scratch at the moment! 

I will update this issue when it is complete.

**Comment by linhduongtuan** [comment]
Hi @lukemelas,
Firstly, I thank you for your interesting codes from the EfficientNet paper using TF to PyTorch.
I have try your code and reimplement to 102_Flowers dataset by using Google Colab with GPU. However, it takes to much time to train the dataset comparing with DenseNet161 (work perfectly).
Could you train the dataset from scratch to test your code?
Sincerely, Linh

**Comment by yuskey** [comment]
I'm more curious about how to apply transfer learning on here. So if I understand the code right, running model.forward() runs the feature extractor and then what would be referred to as the "top" of the network (i.e. the classifier part). If I want to leverage the pretrained weights for the feature extractor and retrain the classifier, would I just load in the weights up to the last _FC linear layer or should I load i the weights up to the feature extractor and retrain the rest?

For reference I am asking if I should retrain this whole block of code below or just the self._fc(x) line:
        x =relu_fn(self._bn1(self._conv_head(x)))
        x = F.adaptive_avg_pool2d(x, 1).squeeze(-1).squeeze(-1)
        if self._dropout:
            x = F.dropout(x, p=self._dropout, training=self.training)
        x = self._fc(x)

**Comment by LinxiFan** [comment]
@lukemelas have you made progress on reproducing the ImageNet training from scratch? My best effort so far is still far below the reported numbers: #66
